myshop/tests/shop/test_urls.py

```diff
@@ -0,0 +1,27 @@
+from django.test import SimpleTestCase
+from django.urls import reverse, resolve
+from shop.views import product_list, product_detail
+from .test_modelmixintestcase import ModelMixinTestCase
+
+
+class TestURLs(ModelMixinTestCase, SimpleTestCase):
+    def test_product_list(self):
+        self.product_list_url = reverse("shop:product_list")

```

Comment:
Use instance variable only when the value is needed throught the class. In this case this is being used only in this function.

Also instance variable should be initialized only in the top level i.e. set up method or init

Reply:
Resolved